### ISSUE TITLE
Change NaCl Signal Handler to shutdown individual cages on faults, not entire Lind system

### ISSUE THREAD

**Issue by rennergade** [Lind-Project/lind_project#210]
Currently when a cage encounters a fault (either in trusted or untrusted) the signal handler will catch that fault and then shutdown NaCl.

Instead we should have the signal handler catch the fault and just shutdown the cage it was caught in.

The handler seems to operate in `void SignalCatch` in `nacl_signal.c`

This change will make it easier to fuzz Lind via trinity, which now is working.

**Comment by rennergade** [comment]
This is done and PR'ed but I'm hoping to add a better logging system for faults so @TristanB22 can easily post bug reports for us to fix.

**Comment by rennergade** [comment]
Most of this has been done for a bit, and I just added in the proper nap teardown feature. I haven't merged this for two reasons.

1. I've had an issue with containers freezing while trying to debug this, and originally I thought there was still something wrong here. I've had that issue elsewhere now as I've tackled other issues and I'm beginning to suspect its the LKM. The issue usually manifests while running Lind in gdb or rr. The process can freeze and become unexitable. Trying to kill that process externally ends up freezing the docker container. And finally nuking the container externally ends up freezing the server. Will be looking into this more now...

2. While trying to debug some of the address translation (EFAULT) stuff, I noticed this branch deadlocks when either a fault occurs in NaCl were locks are held (this is usually the LOG lock), or if a panic happens in SafePOSIX while locks are held (dashmap should help here mostly). Going to see what I can do here to fix this problem and get this all merged.
